src/factory.ts

```diff
@@ -9,9 +9,9 @@ export interface UndoStoreState {
   futureStates: any[];
   isUndoHistoryEnabled: boolean;
 
-  undo: () => void;
-  redo: () => void;
-  clear: () => void;
+  undo: (steps?: number) => void;
+  redo: (steps?: number) => void;
```

Comment:
Now that I'm looking at this, could the parameter of undo and redo be something like `options` that is an object? Right now the object would be `options: { steps?: number }`, but I could see us expanding the options. That way we wouldn't have breaking changes in the future if we want to add more. What do you think?

Reply:
Not sure what other options it would make sense to add to these functions. If you can think of a few important ones than sure, otherwise in the future you could always add a second parameter that accepts an options object. Not having to pass on options object feels more streamlined to me.

Reply:
True, can think of any off-hand. I'm good with keeping this streamlined until there is actually a use for multiple parameters.